ktor-network/jvm/src/io/ktor/network/sockets/DatagramSocketImpl.kt

```diff
@@ -51,6 +52,7 @@ internal class DatagramSocketImpl(
                 channel.send(receiveImpl())
             }
         } catch (_: ClosedChannelException) {
+        } catch (cause: IOException) {
```

Comment:
This can be thrown by `channel.receive(buffer)` in the same file, so the exception must be caught here to prevent the app from crashing. This was already done for native: https://github.com/ktorio/ktor/blob/main/ktor-network/nix/src/io/ktor/network/sockets/DatagramSocketNative.kt#L50